sqlx-postgres/src/types/time/time.rs

```diff
@@ -40,10 +41,21 @@ impl<'r> Decode<'r, Postgres> for Time {
                 Time::MIDNIGHT + Duration::microseconds(us)
             }
 
-            PgValueFormat::Text => Time::parse(
-                value.as_str()?,
-                &format_description!("[hour]:[minute]:[second].[subsecond]"),
-            )?,
+            PgValueFormat::Text => {
+                let s = value.as_str()?;
+
+                // If there is no decimal point we need to add one.
+                let s = if s.contains('.') {
+                    Cow::Borrowed(s)
+                } else {
+                    Cow::Owned(format!("{s}.0"))
+                };
+
+                Time::parse(
+                    &s,
+                    &format_description!("[hour]:[minute]:[second].[subsecond]"),
+                )?
+            }
```

Comment:
Actually the fix is much simpler, just make the subsecond optional in the format:

```suggestion
            PgValueFormat::Text => Time::parse(
                value.as_str()?,
                // Postgres will not include the subsecond part if it's zero.
                &format_description!("[hour]:[minute]:[second][optional [.[subsecond]]]"),
            )?,
```

`Time::parse()` assumes it's zero if not present.

Reply:
Thank you for your review.
I updated the code as you suggested.